src/target/ext/host_io.rs

```diff
@@ -172,6 +172,22 @@ pub enum HostIoError<E> {
     Fatal(E),
 }
 
+/// Converts a `std::io::Error` into a `HostIoError::Errno`.
```

Comment:
nice. this is a good inclusion.

could you also add a comment to the `HostIoError` docs that points out this impl? e.g:

```
/// When the `std` feature is enabled, `HostIoError` implements `From<std::io::Error>`, mapping [`std::io::ErrorKind`] 
/// to the appropriate [`HostIoErrno`] when possible, and falling back to [`HostIoErrno::EUNKNOWN`] when no 
/// mapping exists.
```